code/modules/mob/living/carbon/xenomorph/Xenomorph.dm

```diff
@@ -344,6 +349,16 @@
 		src.hivenumber = old_xeno.hivenumber
 	else if(hivenumber)
 		src.hivenumber = hivenumber
+	//putting the organ in for research
+	if(organ_value != 0)
+		var/obj/item/organ/xeno/organ = new() //give
+		organ.forceMove(src)
+		organ.research_value = organ_value
+		organ.caste_origin = src.caste_type
+		if(!isqueen(src) && !ispredalien(src))
+			organ.icon_state = "heart_t[src.tier]"
+		else
+			organ.icon_state = "heart_t3"
```

Comment:
don't typecheck src ever, instead make a proc on /xeno lke

```dm
/mob/living/carbon/xenomorph/proc/get_organ_icon_state()
	return "heart_t[tier]"
	
/mob/living/carbon/xenomorph/predalien/get_organ_icon_state()
	return "heart_t3"
	
/mob/living/carbon/xenomorph/queen/get_organ_icon_state()
	return "heart_t3"
```

and then here you'd just do

```suggestion
		organ.icon_state = get_organ_icon_state()
```